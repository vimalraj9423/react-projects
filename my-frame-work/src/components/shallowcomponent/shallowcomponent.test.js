
import { configure, shallow ,mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'; 
import React from "react";
import toJson from 'enzyme-to-json';
import ShallowComponent,{StateFull} from "./shallowcomponent";
configure({ adapter: new Adapter() }); 

it("shallow component test",()=>{
    let wrapper = shallow(<ShallowComponent/>);
    expect(wrapper.find('.foo')).toHaveLength(2);
    expect(toJson(wrapper)).toMatchSnapshot();
})


it("shallow component test",()=>{
    let wrapper = mount(<ShallowComponent/>);
    console.log(wrapper.instance()) 
    expect(wrapper.find('.foo')).toHaveLength(2);
    expect(toJson(wrapper)).toMatchSnapshot();
    console.log(wrapper.html())
})
it("statefull",()=>{
    let wrapper = shallow(<StateFull/>);
    console.log(wrapper.instance().state);
    expect(wrapper.instance()).toBeInstanceOf(StateFull)
})