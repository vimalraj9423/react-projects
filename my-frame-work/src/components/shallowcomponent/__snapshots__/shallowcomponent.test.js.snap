// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow component test 1`] = `
<div>
  <div
    className="foo"
  />
  <div
    className="foo"
  >
    <DummyComponent />
  </div>
</div>
`;

exports[`shallow component test 2`] = `
<_default>
  <div>
    <div
      className="foo"
    />
    <div
      className="foo"
    >
      <DummyComponent>
        <div>
          dummy
        </div>
      </DummyComponent>
    </div>
  </div>
</_default>
`;
